apis/cassandra/v1beta1/cassandradatacenter_webhook.go

```diff
@@ -38,6 +39,10 @@ const (
 	k8ssandraPrefix string = "k8ssandra.io"
 )
 
+// Some exceptions to the reserved prefixes above:
+var allowedServiceLabels = []string{"k8ssandra.io/cluster-name", "k8ssandra.io/cluster-namespace"}
```

Comment:
suggestion: It would be nice to have these prefixes as consts (while []string can't be const, the individual items can be).

But alternatively, I was going to suggest that if the CassandraDatacenter object itself has owner reference to another object, it would be allowed to have different set of labels/annotations. But we don't apparently do that in the k8ssandra-operator (and control-plane <-> data-plane issue), so that was not going to work. 

That said, cass-operator does not use k8ssandra.io prefixes for CassandraDatacenter (we only use that prefix in the CassandraTask), so perhaps that check can be removed since k8ssandra 1.x is also deprecated. All the labels/annotations are cassandra.datastax.com in cass-operator.

Reply:
> That said, cass-operator does not use k8ssandra.io prefixes for CassandraDatacenter (we only use that prefix in the CassandraTask), so perhaps that check can be removed since k8ssandra 1.x is also deprecated. All the labels/annotations are cassandra.datastax.com in cass-operator.

Makes sense, and it's an even better way to address the issue. I'll revisit this PR.